static/app/views/settings/project/loaderScript.tsx

```diff
@@ -0,0 +1,124 @@
+import {useState} from 'react';
+
+import {Button} from 'sentry/components/button';
+import EmptyMessage from 'sentry/components/emptyMessage';
+import ExternalLink from 'sentry/components/links/externalLink';
+import LoadingError from 'sentry/components/loadingError';
+import LoadingIndicator from 'sentry/components/loadingIndicator';
+import {Panel, PanelAlert, PanelBody, PanelHeader} from 'sentry/components/panels';
+import {t, tct} from 'sentry/locale';
+import {Organization, Project} from 'sentry/types';
+import {useApiQuery} from 'sentry/utils/queryClient';
+import withOrganization from 'sentry/utils/withOrganization';
+import SettingsPageHeader from 'sentry/views/settings/components/settingsPageHeader';
+import TextBlock from 'sentry/views/settings/components/text/textBlock';
+import {LoaderSettings} from 'sentry/views/settings/project/projectKeys/details/loaderSettings';
+import {ProjectKey} from 'sentry/views/settings/project/projectKeys/types';
+
+export function ProjectLoaderScript({
+  organization,
+  project,
+}: {
+  organization: Organization;
+  project: Project;
+}) {
+  const apiEndpoint = `/projects/${organization.slug}/${project.slug}/keys/`;
+  const [updatedProjectKeys, setUpdatedProjectKeys] = useState<ProjectKey[]>([]);
+
+  const {
+    data: projectKeys,
+    isLoading,
+    error,
+  } = useApiQuery<ProjectKey[]>([apiEndpoint], {
+    staleTime: 0,
+  });
+
+  function updateProjectKey(projectKey: ProjectKey) {
+    const existingPos = updatedProjectKeys.findIndex(key => key.id === projectKey.id);
+    if (existingPos > -1) {
+      const updated = [...updatedProjectKeys];
+      updated[existingPos] = projectKey;
+      setUpdatedProjectKeys(updated);
+    } else {
+      setUpdatedProjectKeys([...updatedProjectKeys, projectKey]);
+    }
```

Comment:
just a suggestion but we could do something like this:

```suggestion
const newUpdatedProjectKeys = existingPos > -1 ? [...updatedProjectKeys].map((updatedProjectKey, index) => {
return index === existingPos ? projectKey : updatedProjectKey
}) : [...updatedProjectKeys, projectKey]

setUpdatedProjectKeys(newUpdatedProjectKeys)
```